### DISCUSSION TITLE
How does “Publishing access” settings relate to 2FA settings?

### DISCUSSION THREAD

**Discussion by detj** [npm/feedback#691] [Registry]
I’m a bit confused on how the settings described in “[Requiring 2FA for package publishing and settings modification](https://docs.npmjs.com/requiring-2fa-for-package-publishing-and-settings-modification)” relate to the settings in “[Configuring two-factor authentication](https://docs.npmjs.com/configuring-two-factor-authentication)”.

![Screenshot 2022-05-19 at 9 20 35 PM](https://user-images.githubusercontent.com/280236/169345215-ff45076a-4783-490a-988b-388362d4e6b8.png)

From the above image, if 2FA is enabled for authorization and publishing for an npm user account, then why does each package have an individual setting for “Publishing access” as can be seen below.

![Screenshot 2022-05-19 at 9 33 17 PM](https://user-images.githubusercontent.com/280236/169346695-b2628a94-481f-447c-815c-2437fe6a26d5.png)

Moreover, the setting defaults to “**Two-factor authentication is not required**”. 
Let’s say I publish 10 packages. And if I want to safeguard publishing by enabling 2FA. Do I need to individually configure each package’s 2FA setting? What if I don’t? Are my package publishing setting somehow insecure, even if I have turned on 2FA in account settings?

**Comment by ljharb** [comment]
Yes. Your account 2FA only governs *you*. If you add somebody else to a package, and the package isn't set to require 2FA to publish, then they can publish without having 2FA enabled on *their* account.

In other words, yes, you do currently have to go manually require 2FA on every new package you publish, individually.

**Reply by detj** [reply]
Understood, thanks.
